src/adapter/src/catalog.rs

```diff
@@ -1846,6 +1855,42 @@ impl CatalogItem {
             | CatalogItem::Connection(_) => None,
         }
     }
+
+    pub fn initial_logical_compaction_window(&self) -> Option<Duration> {
+        let custom_logical_compaction_window = match self {
+            CatalogItem::Table(table) => table.custom_logical_compaction_window,
+            CatalogItem::Source(source) => source.custom_logical_compaction_window,
+            CatalogItem::Index(_) => None,
+            CatalogItem::MaterializedView(_) => None,
+            CatalogItem::Log(_)
+            | CatalogItem::View(_)
+            | CatalogItem::Sink(_)
+            | CatalogItem::Type(_)
+            | CatalogItem::Func(_)
+            | CatalogItem::Secret(_)
+            | CatalogItem::Connection(_) => return None,
+        };
+        Some(custom_logical_compaction_window.unwrap_or(DEFAULT_LOGICAL_COMPACTION_WINDOW))
```

Comment:
Does this function always return `Some`? If so, should it just have a return type of `Duration`?

Reply:
Nevermind, I see right there that it can return `None`